Description: Link vlc-cache-gen to gobject to prevent crashes related to gobject
 If plugins use gobject in some way vlc-cache-gen might crash if gobject is
 unloaded and loaded in an unsupported way. Linking vlc-cache-gen makes sure
 that gobject is not unloaded during the run and prevents the crash.
Author: Sebastian Ramacher <sramacher@debian.org>
Bug-Debian: https://bugs.debian.org/752544
Forwarded: https://trac.videolan.org/vlc/ticket/12039
Last-Update: 2014-08-27

Index: vlc-2.2.0~pre2/bin/Makefile.am
===================================================================
--- vlc-2.2.0~pre2.orig/bin/Makefile.am
+++ vlc-2.2.0~pre2/bin/Makefile.am
@@ -72,6 +72,10 @@ vlc_cache_gen_LDADD += vlc_win32_rc.$(OB
 vlc_cache_gen_DEPENDENCIES = vlc_win32_rc.$(OBJEXT)
 endif
 
+if HAVE_GOBJECT
+vlc_cache_gen_LDADD += $(GOBJECT_LIBS)
+endif
+
 #
 # Plug-ins cache
 #
Index: vlc-2.2.0~pre2/configure.ac
===================================================================
--- vlc-2.2.0~pre2.orig/configure.ac
+++ vlc-2.2.0~pre2/configure.ac
@@ -788,6 +788,12 @@ AS_IF([test -n "${PKG_CONFIG}" ],[
     ])
 ])
 
+dnl
+dnl Check for gobject
+dnl
+have_gobject="no"
+PKG_CHECK_MODULES([GOBJECT], [gobject-2.0], [ have_gobject="yes" ])
+AM_CONDITIONAL(HAVE_GOBJECT, [ test "${have_gobject}" = "yes" ])
 
 dnl
 dnl Check for zlib.h and -lz along with system -lminizip if available
